### PR TITLE
Updates to pyflyte init templating

### PR DESCRIPTION
**Pull request by zeryx**
# TL;DR
This PR removes the cookiecutter templating system for `pyflyte init`, and instead uses gitpython, without templating.
This is part of a wider effort to replace our Dockerfile / docker_build.sh systems with the much simpler `ImageSpec`.

This PR depends on https://github.com/flyteorg/flytekit-python-template/pull/40

**This is a breaking change**
All templates are now imagespec based, and cookiecutter is no longer used to generate custom readme titles or directory names.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As discussed as part of the larger Flyte example improvement with newly introduced imagespec.

### CODE REVIEW

File: flytekit/clis/sdk_in_container/helpers.py [hunk 1]
```diff
@@ -63,3 +66,34 @@ def patch_image_config(config_file: Optional[str], image_config: ImageConfig) ->
...
+    - github_url: The URL of the GitHub repository.
+    - branch: The branch of the repository.
+    - src_dir: The directory in the repository to copy.
+    - dest_dir: The local directory to copy into.
+    """
+    # Clone the repo
+    repo = Repo.clone_from(git_url, "temp_repo")
+
+    # Checkout to the branch
+    repo.git.checkout(branch)
```
Comment (eapolinario) [inline flytekit/clis/sdk_in_container/helpers.py:85]: you can pass the branch name directly in the call to `Repo.clone_from`.

File: flytekit/clis/sdk_in_container/helpers.py [hunk 2]
```diff
@@ -63,3 +66,34 @@ def patch_image_config(config_file: Optional[str], image_config: ImageConfig) ->
...
+    # Check if source directory exists
+    if not os.path.exists(src_path):
+        print(f"Template named {src_dir} does not exist in the repository.")
+        return
+
+    # Copy the files from the source directory to the destination directory
+    shutil.copytree(src_path, dest_dir)
+
+    # Remove the temporary cloned repo
+    shutil.rmtree("temp_repo")
```
Comment (eapolinario) [inline flytekit/clis/sdk_in_container/helpers.py:99]: how about we use a [`TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) to hold the directory? This way we don't need to do this book-keeping and it also allows for this command to be run multiple times.

File: setup.py [hunk 3]
```diff
@@ -68,7 +68,6 @@
         "docstring-parser>=0.9.0",
         "diskcache>=5.2.1",
         "cloudpickle>=2.0.0",
-        "cookiecutter>=1.7.3",
```
Comment (eapolinario) [inline setup.py:71]: ❤️

**Review by eapolinario — COMMENTED** [review]
This is awesome!

We might think about using a brand new repo to hold the examples (instead of flytekit-python-template).

**Comment by kumare3** [comment]
cc @eapolinario / @neverett do you think you want to check this one in?
